Core/MIPS/IR/IRRegCache.cpp

```diff
@@ -0,0 +1,61 @@
+#include "Core/MIPS/IR/IRRegCache.h"
+#include "Core/MIPS/IR/IRInst.h"
+
+void IRRegCache::Flush(int rd) {
+	if (rd == 0) {
+		return;
+	}
+	if (reg_[rd].isImm) {
+		ir_->WriteSetConstant(rd, reg_[rd].immVal);
+		reg_[rd].isImm = false;
+	}
+}
+
+void IRRegCache::Discard(int rd) {
+	if (rd == 0) {
+		return;
+	}
+	reg_[rd].isImm = false;
+}
+
+IRRegCache::IRRegCache(IRWriter *ir) : ir_(ir) {
+	memset(&reg_, 0, sizeof(reg_));
+	reg_[0].isImm = true;
+	ir_ = ir;
+}
+
+void IRRegCache::FlushAll() {
+	for (int i = 0; i < TOTAL_MAPPABLE_MIPSREGS; i++) {
+		//if (i < IRTEMP_0)
```

Comment:
Those can be constants too right?  I think we can just remove this line.

-[Unknown]